packages/app/src/components/DonateComponent.tsx

```diff
@@ -51,26 +51,39 @@ function DonateComponent({ collective }: DonateComponentProps) {
     navigate(`/profile/${address}`);
   }
 
-  const [currency, setCurrency] = useState<string>('G$');
   const [frequency, setFrequency] = useState<Frequency>(Frequency.Monthly);
   const [duration, setDuration] = useState(12);
   const [decimalDonationAmount, setDecimalDonationAmount] = useState(0);
 
   const tokenList = useTokenList();
+  const gdEnvSymbol =
+    Object.keys(tokenList).find((key) => {
+      if (key.startsWith('G$')) {
+        return tokenList[key].address.toLowerCase() === collective.rewardToken.toLowerCase();
+      } else return false;
+    }) || 'G$';
+
+  const GDToken = GDEnvTokens[gdEnvSymbol];
+
   const currencyOptions: { value: string; label: string }[] = useMemo(() => {
-    let options = Object.keys(tokenList).map((key) => ({
-      value: key,
-      label: key,
-    }));
+    let options = Object.keys(tokenList)
+      .filter((key) => key.startsWith('G$') === false || key === gdEnvSymbol)
+      .map((key) => ({
+        value: key,
+        label: key,
+      }));
```

Comment:
```suggestion
    let options = Object.keys(tokenList).reduce((acc, key) => {
  if (!key.startsWith('G$') || key === gdEnvSymbol) {
    acc.push({ value: key, label: key });
  }
  return acc;
}, []);
```

Reduce I find personally more readable.
and it also only requires a single loop